EFCore/MAUI/Readme.md

```diff
@@ -0,0 +1,873 @@
+
+### Implement and test first views: "Login" and "Item List"
+
+1. Register the `WebAPIService` as well as routes and initial navigation in `App.xaml.cs`.
+
+   _App.xaml.cs_:
+   ```cs
+   public App() {
+      InitializeComponent();
+      DependencyService.Register<NavigationService>();
+      DependencyService.Register<WebAPIService>();
+
+      Routing.RegisterRoute(typeof(ItemsPage).FullName, typeof(ItemsPage));
+   
+      MainPage = new MainPage();
+      var navigationService = DependencyService.Get<INavigationService>();
+      navigationService.NavigateToAsync<LoginViewModel>(true);
+   }
+   ```
+
+2. Modify the `LoginViewModel` to authenticate in the Web API service and navigate to the Items page after successful authentication.
+
+   _ViewModels/LoginViewModel.cs_:
+   ```cs
+   async void OnLoginClicked() {
+      IsAuthInProcess = true;
+      var response = await ((WebAPIService)DataStore).Authenticate(userName, password);
+      IsAuthInProcess = false;
+      if (!string.IsNullOrEmpty(response)) {
+          ErrorText = response;
+          HasError = true;
+          return;
+      }
+      HasError = false;
+      await Navigation.NavigateToAsync<ItemsViewModel>();
+   }
+
+   ```
+
+3. In the `ItemsViewModel`, modify the `ExecuteLoadItemsCommand` to request all posts from the `IDataStore` service.
+
+   _ViewModels/ItemsViewModel.cs_:
+   ```cs
+   async Task ExecuteLoadItemsCommand() {
+      IsBusy = true;
+      try
+      {
+         Items.Clear();
+         var items = await DataStore.GetItemsAsync(true);
+         foreach (var item in items) {
+            Items.Add(item);
+         }
+      }
+      catch (Exception ex) {
+         System.Diagnostics.Debug.WriteLine(ex);
+      }
+      finally {
+         IsBusy = false;
+      }
+   }  
+   ```
+
+Start the Web API project, and then the .NET MAUI app. Log in as an Editor or Viewer user created in the Web API Module Updater.
+
+![](../../images/MAUI/LogIn.png)
+
+After you log in, you will see the predefined posts displayed in a list view.
+
+![](../../images/MAUI/CollectionView.png)
+
+## Extend Backend and Frontend Apps
+
+### Add an "Item Details" view
+
+1. Implement a Web API service endpoint that serves the author's photo for a post based on the post's ID. 
+
+   First, add the `Photo` property to the `ApplicationUser` persistent class:
+
+   _BusinessObjects/ApplicationUser.cs_:
+   ```cs
+   public class ApplicationUser : PermissionPolicyUser, ISecurityUserWithLoginInfo {
+      // ...
+      public virtual MediaDataObject Photo { get; set; }
+   }
+   ```
+
+   Add a logic that assigns photos to predefined users to the Module Updater:
```

Comment:
You say "add logic .. to the Module Updater", and there's a lot of stuff in between, which makes it a bit hard to read. (The fact that you repeat the preposition "to" contributes to the confusion.) I'm changing the flow a bit, so that related things go together. 

"Logic" is a mass noun and doesn't require an article.

```suggestion
   Modify the Module Updater. Add logic that assigns photos to predefined users:
```